exec_d.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package libcnb
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb/internal"
+)
+
+//go:generate mockery --name Executor --case=underscore
+
+// Executor describes an interface for types that follow the Exec.d specification.
+// It should return a map of environment variables and their values as output.
+type Executor interface {
```

Comment:
Or possibly having the interface be `ExecD`, which lines up with the spec pretty well too, and having the `ExecD` function be something different like `InvokeExecD` or `RunExecD`? Since the primary purpose of that function is to trigger the setup & execution of an ExecD executable.

Reply:
Not too tied to the existing names. I was also struggling with the appropriate names for these. I like the latter idea since it makes it clear what's happening but also okay with the first one.